hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/DataTieringManager.java

```diff
@@ -61,25 +63,25 @@ private DataTieringManager(Map<String, HRegion> onlineRegions) {
    * Initializes the DataTieringManager instance with the provided map of online regions.
    * @param onlineRegions A map containing online regions.
    */
-  public static synchronized void instantiate(Map<String, HRegion> onlineRegions) {
-    if (instance == null) {
-      instance = new DataTieringManager(onlineRegions);
-      LOG.info("DataTieringManager instantiated successfully.");
+  public static synchronized void instantiate(Configuration conf,
```

Comment:
Why do we need this Configuration parameter? We only enter here when isDataTieringFeatureEnabled(conf) is true, so why do we need to check it again?

Reply:
This is an additional check to avoid an unintentional use if someone makes a call to this function.

Reply:
I have made the function isDataTieringFeatureEnabled private function of the class. We do not want to give the undue flexiblity to the user to instantiate the DataTieringManager even when the feature key is disabled.

Reply:
Also, would it make sense to return True if instantiated successfully and false otherwise?